Feature | ER-18 Register user with organization creation and owner permissions

Thushana24 opened the pull request:
## Register user with organization creation and owner permissions

https://kifgo.atlassian.net/browse/ER-18

## Type of Change

- [X] New feature  
- [ ] Bug fix  
- [ ] Refactoring  
- [ ] Security patch  
- [ ] UI/UX improvement  

## Description

Implemented organization creation and owner role assignment during user registration.  
Introduced centralized permission management for the owner role through a new `permission.ts` file.

## What is fixed / implemented?

- Extend user registration to:
  - Create a new organization.
  - Connect the registered user as the organization owner.
  - Insert an organization member entry with the `OWNER` role and assigned permissions.
- Import `UserRole` from Prisma Client for type safety.
- Add `permission.ts` defining `OWNER_PERMISSIONS`.
- Wrap operations inside a Prisma transaction for data consistency.

## Additional Information

_N/A_

## Screenshots (if applicable)

_N/A_

## Checklist

- [ ] I have added or updated relevant tests to cover the changes.  
- [X] I have performed a self-review of my code.  
- [X] My changes generate no new warnings or errors in development and production builds.  
- [ ] I have verified the changes on multiple devices and browsers (if applicable).  
- [ ] Environment variables have been updated (if applicable).  
- [ ] New packages have been installed and documented (if applicable).

sabesh82 committed:
feat: add user and organization validation schemas

- Implement Zod schemas for user registration and login
- Add organization name validation schema
- Update create-diff:win script in package.json to improve diff output

Aathil1681 committed:
feat(api): add user and organization existence checks to register route

- Validates incoming request using `RegisterUserSchema`
- Checks if a user with the given email already exists
- Checks if an organization with the given name already exists
- Returns appropriate 409 conflict responses with error codes and messages

sujeevanan committed:
Add user registration with password hashing using argon2

- Added argon2 and @types/argon2 dependencies to package.json
- Updated register route to hash passwords with argon2
- Implemented user creation with validated data in a Prisma transaction
- Included checks for existing users

Thushana24 committed:
feat(auth): register user along with organization creation and owner permissions

- Import UserRole from Prisma and OWNER_PERMISSIONS
- Include organization creation and organization member entry inside transaction
- Fix destructuring to include organizationName
- Add permission.ts to define OWNER_PERMISSIONS

Dhuvii commented on src/app/permission.ts:
```diff
@@ -0,0 +1,6 @@
+export const OWNER_PERMISSIONS: string[] = [
+    "ORGANIZATION::", // Owners can do anything with all organizations
```
```
export const OWNER_PERMISSIONS = [
  "ORGANIZATION:*:*", // Owners can do anything with all organizations
  "USER:*:*", // Owners can do anything with all users
  "RATING:*:*", // Owners can do anything with all ratings
];
```

Dhuvii commented on src/app/api/auth/register/route.ts:
```diff
@@ -0,0 +1,84 @@
+            Owner: {
+                connect: {
+                    id: user.id,
+                    
```
Remove this extra space

Thushana24 committed:
refactor(permission, auth): update owner permissions format and clean up register route

- Updated OWNER_PERMISSIONS to use 'RESOURCE:*:*' wildcard pattern for consistency.
- Removed unnecessary whitespace in the register route.
- Added VSCode workspace setting for Next.js extension prompt.
